synapse/rest/client/v2_alpha/user_directory.py

```diff
@@ -59,6 +59,12 @@ def on_POST(self, request):
         requester = yield self.auth.get_user_by_req(request, allow_guest=False)
         user_id = requester.user.to_string()
 
+        if not self.hs.config.user_directory_search_enabled:
```

Comment:
I haven't moved that into `__init__` as there's no way of getting at servlets after they're set up afaik. We could do various contortions in the test to make it work, but I don't know whether that is better or worse.